transactions/transactions.go

```diff
@@ -12,6 +13,74 @@ import (
 
 const maxGasLimit = 9999
 
+type SignedTransaction struct {
+	flow.Transaction
+}
+
+// Signatures JSON HTTP response
+type SignedTransactionJSONResponse struct {
+	Code               string                     `json:"code"`
+	Arguments          [][]byte                   `json:"arguments"`
+	ReferenceBlockID   string                     `json:"referenceBlockId"`
+	GasLimit           uint64                     `json:"gasLimit"`
+	ProposalKey        ProposalKeyJSON            `json:"proposalKey"`
+	Payer              string                     `json:"payer"`
+	Authorizers        []string                   `json:"authorizers"`
+	PayloadSignatures  []TransactionSignatureJSON `json:"payloadSignatures"`
+	EnvelopeSignatures []TransactionSignatureJSON `json:"envelopeSignatures"`
+}
+
+type ProposalKeyJSON struct {
+	Address        string `json:"address"`
+	KeyIndex       int    `json:"keyIndex"`
+	SequenceNumber uint64 `json:"sequenceNumber"`
+}
+
+type TransactionSignatureJSON struct {
+	Address   string `json:"address"`
+	KeyIndex  int    `json:"keyIndex"`
+	Signature string `json:"signature"`
+}
+
+func (st *SignedTransaction) ToJSONResponse() SignedTransactionJSONResponse {
+	var res SignedTransactionJSONResponse
+
+	res.Code = string(st.Script)
+	res.Arguments = st.Arguments
```

Comment:
This currently results as an array of base64 encoded arguments. Is that ok or should these types be de-structured and serialized into JSON one step further?

Reply:
Actually, this will improve in following PR (see [specific commit here](https://github.com/tuommaki/flow-wallet-api/pull/1/commits/e536ce9115ce160c0986638b293a89d548e9ef28)). The base64 encoding is slightly fragile in different use cases due to trailing line feeds etc.